signals.go

```diff
@@ -45,6 +46,10 @@ func serverSignalHandler(listener net.Listener, statusListener net.Listener, sto
 			switch sig {
 			case syscall.SIGTERM:
 				logger.Printf("received SIGTERM, stopping listener")
+				time.AfterFunc(5*time.Minute, func() {
```

Comment:
@nerdrew Yes, it's 5 minutes. I'll just make it a flag though based on @mcpherrinm's suggestion.